nolearn/lasagne/base.py

```diff
@@ -582,30 +582,34 @@ def train_loop(self, X, y, epochs=None):
 
             t0 = time()
 
+            batch_train_sizes = []
             for Xb, yb in self.batch_iterator_train(X_train, y_train):
                 batch_train_loss = self.apply_batch_func(
                     self.train_iter_, Xb, yb)
-                train_losses.append(batch_train_loss)
+                train_losses.append(batch_train_loss[0])
```

Comment:
Good catch – I forgot to mention this in the description.

It was because `batch_train_loss` is actually a list. See https://github.com/dnouri/nolearn/blob/master/nolearn/lasagne/base.py#L502. I am not sure why `loss_train` is wrapped in a list there.